packages/opentelemetry-node/README.md

```diff
@@ -92,6 +92,14 @@ provider.register()
 ## Examples
 See how to automatically instrument [http](https://github.com/open-telemetry/opentelemetry-js/tree/master/examples/http) and [gRPC](https://github.com/open-telemetry/opentelemetry-js/tree/master/examples/grpc) using node-sdk.
 
+## Experimental API methods
+
+#### withSpanAsync
+
+The `NodeTracer` expose a method called `withSpanAsync` that allows you to set a current span in the context of a `async` function. This previously was a problem with `withSpan` because its synchronous behavior, read about that in [this issue](https://github.com/open-telemetry/opentelemetry-js/issues/752).
```

Comment:
Could this functionality be integrated into the existing `withSpan` method by detecting if a promise has been passed back? Not sure if I missed the discussion around this while catching up, but the ergonomics would be better if you didn't have to change the method call when you swap a callback in `withSpan` to an async function.

You should be able to call `fn`, and clear the context in a `.finally` (or `then/catch` pair) off the return value. I would imagine the context propagation for the interstitial promises could be handled by async_hooks.

Reply:
There has been a lot of discussion around this and we decided to keep them separate for now. This decision came down to performance impact and the higher likelihood for subtle bugs using the more complex async mechanism.